providers/implementations/digests/blake2_prov.c

```diff
@@ -35,7 +35,75 @@ IMPLEMENT_digest_functions(blake2s256, BLAKE2S_CTX,
                            ossl_blake2s_final)
 
 /* ossl_blake2b512_functions */
-IMPLEMENT_digest_functions(blake2b512, BLAKE2B_CTX,
-                           BLAKE2B_BLOCKBYTES, BLAKE2B_DIGEST_LENGTH, 0,
-                           ossl_blake2b512_init, ossl_blake2b_update,
-                           ossl_blake2b_final)
+
+static OSSL_FUNC_digest_init_fn blake2b512_internal_init;
+static OSSL_FUNC_digest_newctx_fn blake2b512_newctx;
+static OSSL_FUNC_digest_freectx_fn blake2b512_freectx;
+static OSSL_FUNC_digest_dupctx_fn blake2b512_dupctx;
+static OSSL_FUNC_digest_final_fn blake2b512_internal_final;
```

Comment:
I would expect to also see forward declarations for
`blake2b512_get_params`, `ossl_blake2b_set_ctx_params`, `ossl_blake2b_settable_ctx_params` as well.

Reply:
`blake2b512_get_params` was present elsewhere in the file, moved up

`ossl_blake2b_set_ctx_params` and `ossl_blake2b_settable_ctx_params` are declared in `providers/implementations/include/prov/blake2.h` that gets included. Would you prefer to include a forward declaration in the file anyway?